src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -674,14 +674,14 @@ internal static void CreateHardLink(string path, string pathToTarget)
 
         internal static FileSystemInfo? ResolveLinkTarget(string linkPath, bool returnFinalTarget, bool isDirectory)
         {
-            ValueStringBuilder sb = new(Interop.DefaultPathBufferSize);
+            ValueStringBuilder sb = new(stackalloc char[Interop.DefaultPathBufferSize]);
```

Comment:
Why isn't `using` appropriate here and avoid the `Dispose()` placement issues below?